### ISSUE TITLE
`*': negative argument 

### ISSUE THREAD

**Issue by Flipez** [piotrmurach/tty-box#7]
### Describe the problem
With a certain body the box fails to draw.

### Steps to reproduce the problem
```
        box = TTY::Box.frame(
          width: TTY::Screen.width,
          height: mr.description.lines.count + 10,
          padding: 1,
          title: { top_center: mr.title[0..TTY::Screen.width - 3],
                   bottom_left: "State: #{mr.state}",
                   bottom_right: "by #{mr.author.name}" }
        ) do
          mr.description # content is "Closes #360\r\n\r\nCloses !217 "
        end
        puts box
```

### Actual behaviour
```
Traceback (most recent call last):
        16: from /usr/local/bin/gl:23:in `<main>'
	15: from /usr/local/bin/gl:23:in `load'
	14: from /var/lib/gems/2.5.0/gems/gl-0.3.1/exe/gl:7:in `<top (required)>'
	13: from /var/lib/gems/2.5.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	12: from /var/lib/gems/2.5.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	11: from /var/lib/gems/2.5.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	10: from /var/lib/gems/2.5.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	 9: from /var/lib/gems/2.5.0/gems/thor-0.20.3/lib/thor.rb:238:in `block in subcommand'
	 8: from /var/lib/gems/2.5.0/gems/thor-0.20.3/lib/thor/invocation.rb:115:in `invoke'
	 7: from /var/lib/gems/2.5.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	 6: from /var/lib/gems/2.5.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	 5: from /var/lib/gems/2.5.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	 4: from /var/lib/gems/2.5.0/gems/gl-0.3.1/lib/gl/cli/merge_requests.rb:11:in `list'
	 3: from /var/lib/gems/2.5.0/gems/tty-box-0.4.0/lib/tty/box.rb:100:in `frame'
	 2: from /var/lib/gems/2.5.0/gems/tty-box-0.4.0/lib/tty/box.rb:100:in `times'
	 1: from /var/lib/gems/2.5.0/gems/tty-box-0.4.0/lib/tty/box.rb:112:in `block in frame'
/var/lib/gems/2.5.0/gems/tty-box-0.4.0/lib/tty/box.rb:112:in `*': negative argument (ArgumentError)

```

### Describe your environment

* OS version: Ubuntu 18.04.3
* Ruby version: 2.5.1p57
* TTY::Box version: 0.4.0

**Comment by piotrmurach** [comment]
Thanks for reporting! Do you have time to fix? If not, can you submit a failing test?

**Comment by Flipez** [comment]
Hi, I'll try to give it a look in the next days :)

**Comment by piotrmurach** [comment]
One thing I would suggest is to create a test example that reproduces the issue clearly and independently from any other variables etc... For example, I don't know what `mr` is and what are the `title`, `state`, `author` and `description` exact values. It would be rather hard to figure out whether your data assumptions are a bit off or the actual library is broken. If you don't have time then at least please provide an example with actual values.

**Comment by Flipez** [comment]
I've created a spec with the following failing box:

```ruby
  it 'it renders the box content' do
    box = TTY::Box.frame(
      width: 27,
      height: 7
    ) do
      "Closes #360\r\n\r\nCloses !217"
    end

    expect(box).to eq([
      "┌──────────────────────────┐\n",
      "│Closes #360               │\n",
      "│                          │\n",
      "│Closes !217               │\n",
      "│                          │\n",
      "│                          │\n",
      "└──────────────────────────┘\n"
    ].join)
  end
```

If `width` is `28` everything works.

**Comment by piotrmurach** [comment]
This wasn't as "obvious" as expected. What you found was the result of the issue but not the reason. The reason this was broken is due to the `\r\n` line ending, which is usually found on Windows machines. So I presume this is something you've encountered whilst working on windows. On Unix systems line endings are encoded as `\n`. It is enough to replace `\r\n` with `\n` to fix the problem.

The core issue was that the `strings` dependency when encountering `\r\n` would break the string into lines based only on `\n` character and treat the return char `\r` as a normal character even though it is not printable. Hence the difference between sizes!

I changed [strings](https://github.com/piotrmurach/strings/commit/a61e222101dacfa6a1346c130a217cd0f3f9d21c) to be smart about what line breaks are used and take it into account when tokenizing strings.

Released `v0.4.1` with the fix.
